lib/filesystem.nix

```diff
@@ -306,52 +306,65 @@ in
         As a result, directories with no `.nix` files (including empty
         directories) will be transformed into empty attribute sets.
 
-    # Inputs
-
-    Structured function argument
-
-    : Attribute set containing the following attributes.
-      Additional attributes are ignored.
-
-      `callPackage`
-
-      : `pkgs.callPackage`
-
-        Type: `Path -> AttrSet -> a`
-
-      `directory`
+    # Type
 
-      : The directory to read package files from
+    ```
+    packagesFromDirectoryRecursive :: {
+      callPackage :: Path -> {} -> a,
+      directory :: Path,
+      ...
+    } -> AttrSet
+    ```
 
-        Type: `Path`
+    # Inputs
 
+    `callPackage`
+    : The function used to convert a nix file's path, into a leaf of the attrset.
+      It is typically the `callPackage` function, taken from either `pkgs` or a new scope corresponding to the `directory`.
 
-    # Type
+    `directory`
+    : The directory to read package files from.
 
-    ```
-    packagesFromDirectoryRecursive :: AttrSet -> AttrSet
-    ```
 
     # Examples
     :::{.example}
-    ## `lib.filesystem.packagesFromDirectoryRecursive` usage example
+    ## Basic use of `lib.packagesFromDirectoryRecursive`
 
     ```nix
     packagesFromDirectoryRecursive {
       inherit (pkgs) callPackage;
       directory = ./my-packages;
     }
     => { ... }
+    ```
+
+    In this case, `callPackage` will only search `pkgs` for a file's input parameters.
+    In other words, a file cannot refer to another file in the directory in its input parameters.
+    :::
 
+    :::{.example}
+    ## Create a scope for the nix files found in a directory
+    ```nix
     lib.makeScope pkgs.newScope (
       self: packagesFromDirectoryRecursive {
-        callPackage = self.callPackage;
+        inherit (self) callPackage;
         directory = ./my-packages;
       }
     )
     => { ... }
     ```
 
+    Here, if `directory` contains `awoo.nix` and `owo/uwu.nix`, with the latter expecting `{ awoo }` as a parameter,
```

Comment:
Can you make up better directory and file names. Maybe just copy a simple real world example. I find those names a bit confusing.

Reply:
If the other ones you found use 'a' and 'b' i would like you to stick to that. Its more simple and lets the Reader focus on what is important